[bot\command_modules\game_setup_cmds.py] - 19 issue(s)
  Line 29: Type "Client" is not assignable to declared type "RPGBot"
  Line 55: Cannot access attribute "get_master_role_id" for class "GameManager"
  Line 69: Type "Client" is not assignable to declared type "RPGBot"
  Line 81: Cannot access attribute "get_gm_channel_id" for class "GameManager"
  Line 148: Cannot access attribute "execute" for class "object"
  Line 176: Argument of type "object" cannot be assigned to parameter "db_session" of type "AsyncSession" in function "create_entity"
  Line 184: Cannot access attribute "commit" for class "object"
  Line 191: "session" is possibly unbound
  Line 191: Cannot access attribute "is_active" for class "object"
  Line 192: "session" is possibly unbound
  Line 192: Cannot access attribute "rollback" for class "object"
  Line 198: "session" is possibly unbound
  Line 198: Cannot access attribute "is_active" for class "object"
  Line 199: "session" is possibly unbound
  Line 199: Cannot access attribute "rollback" for class "object"
  Line 227: Argument of type "str | Any | None" cannot be assigned to parameter "language" of type "str" in function "create_new_character"
  Line 247: Argument of type "str | Any | None" cannot be assigned to parameter "key" of type "str" in function "get"
  Line 247: No overloads for "get" match the provided arguments
  Line 514: Argument of type "Any | None" cannot be assigned to parameter "key" of type "str" in function "__setitem__"

[bot\command_modules\inventory_cmds.py] - 19 issue(s)
  Line 115: Expected 3 positional arguments
  Line 126: Expected 3 positional arguments
  Line 128: Expected 3 positional arguments
  Line 132: Argument of type "NLUDataService" cannot be assigned to parameter "game_log_manager" of type "GameLogManager | None" in function "parse_player_action"
  Line 139: Expected 3 positional arguments
  Line 144: Expected 3 positional arguments
  Line 152: Expected 3 positional arguments
  Line 155: Expected 3 positional arguments
  Line 166: Expected 3 positional arguments
  Line 174: Cannot access attribute "get_core_rules_config_for_guild" for class "RuleEngine"
  Line 180: Expected 3 positional arguments
  Line 182: Argument of type "NLUDataService" cannot be assigned to parameter "game_log_manager" of type "GameLogManager | None" in function "parse_player_action"
  Line 189: Expected 3 positional arguments
  Line 203: Expected 3 positional arguments
  Line 211: Cannot access attribute "get_core_rules_config_for_guild" for class "RuleEngine"
  Line 217: Expected 3 positional arguments
  Line 224: Argument of type "NLUDataService" cannot be assigned to parameter "game_log_manager" of type "GameLogManager | None" in function "parse_player_action"
  Line 229: Expected 3 positional arguments
  Line 231: "item_template_id_to_unequip" is not defined

[bot\game\services\consequence_processor.py] - 19 issue(s)
  Line 124: Cannot access attribute "modify_npc_stats" for class "NpcManager"
  Line 140: Cannot access attribute "modify_character_stats" for class "CharacterManager"
  Line 159: Cannot access attribute "give_item_to_character" for class "ItemManager"
  Line 180: Cannot access attribute "notify_player" for class "NotificationService"
  Line 195: Cannot access attribute "remove_item_from_character_by_instance_id" for class "ItemManager"
  Line 197: Cannot access attribute "remove_item_from_character_by_template_id" for class "ItemManager"
  Line 208: Cannot access attribute "notify_player" for class "NotificationService"
  Line 223: Cannot access attribute "move_character_to_location" for class "CharacterManager"
  Line 226: Cannot access attribute "move_npc_to_location" for class "NpcManager"
  Line 246: "Character" is not awaitable
  Line 246: "None" is not awaitable
  Line 250: Cannot access attribute "start_event" for class "EventManager"
  Line 270: Cannot access attribute "update_quest_progress" for class "QuestManager"
  Line 287: Cannot access attribute "notify_player" for class "NotificationService"
  Line 300: Cannot access attribute "apply_status_effect" for class "StatusManager"
  Line 350: "Character" is not awaitable
  Line 350: "None" is not awaitable
  Line 352: Cannot access attribute "award_experience" for class "RuleEngine"
  Line 366: Cannot access attribute "notify_player" for class "NotificationService"

[bot\game\turn_processor.py] - 19 issue(s)
  Line 58: Argument of type "object" cannot be assigned to parameter "db_session" of type "AsyncSession" in function "get_entities"
  Line 78: Argument of type "List[Dict[str, Any]]" cannot be assigned to parameter "s" of type "str | bytes | bytearray" in function "loads"
  Line 87: Cannot access attribute "analyze_actions_for_conflicts" for class "ConflictResolver"
  Line 110: Cannot access attribute "add" for class "object"
  Line 142: Cannot access attribute "get_player_account_by_char_id" for class "CharacterManager"
  Line 153: Argument of type "List[Dict[str, Any]]" cannot be assigned to parameter "s" of type "str | bytes | bytearray" in function "loads"
  Line 185: Cannot access attribute "character_action_processor" for class "GameManager"
  Line 188: Cannot access attribute "send_character_feedback" for class "NotificationService"
  Line 228: No parameter named "session"
  Line 234: Cannot access attribute "character_action_processor" for class "GameManager"
  Line 245: Cannot access attribute "send_character_feedback" for class "NotificationService"
  Line 249: Cannot access attribute "send_character_feedback" for class "NotificationService"
  Line 263: Cannot access attribute "send_character_feedback" for class "NotificationService"
  Line 273: Cannot assign to attribute "collected_actions_json" for class "Character"
  Line 274: Cannot access attribute "add" for class "object"
  Line 284: Cannot assign to attribute "collected_actions_json" for class "Character"
  Line 285: Cannot access attribute "add" for class "object"
  Line 290: Cannot access attribute "commit" for class "object"
  Line 296: Cannot access attribute "rollback" for class "object"

[tests\ai\test_ai_response_validator.py] - 19 issue(s)
  Line 68: Cannot access attribute "general_settings" for class "CoreGameRulesConfig"
  Line 69: Cannot access attribute "general_settings" for class "CoreGameRulesConfig"
  Line 73: Cannot access attribute "character_stats_rules" for class "CoreGameRulesConfig"
  Line 75: Cannot access attribute "character_stats_rules" for class "CoreGameRulesConfig"
  Line 77: Cannot access attribute "item_rules" for class "CoreGameRulesConfig"
  Line 125: Object of type "None" is not subscriptable
  Line 143: Object of type "None" cannot be used as iterable value
  Line 144: Object of type "None" cannot be used as iterable value
  Line 145: Object of type "None" cannot be used as iterable value
  Line 146: Object of type "None" cannot be used as iterable value
  Line 214: Object of type "None" cannot be used as iterable value
  Line 215: Object of type "None" cannot be used as iterable value
  Line 233: Object of type "None" cannot be used as iterable value
  Line 234: Object of type "None" cannot be used as iterable value
  Line 253: Object of type "None" cannot be used as iterable value
  Line 254: Object of type "None" cannot be used as iterable value
  Line 255: Object of type "None" cannot be used as iterable value
  Line 288: Object of type "None" cannot be used as iterable value
  Line 304: Object of type "None" cannot be used as iterable value

[bot\game\managers\character_manager.py] - 18 issue(s)
  Line 33: "NPCManager" is unknown import symbol
  Line 109: Argument of type "Character" cannot be assigned to parameter "entity" of type "Player | NPC | Character" in function "calculate_effective_stats"
  Line 114: Cannot assign to attribute "effective_stats_json" for class "Character"
  Line 171: Cannot access attribute "from_db_model" for class "type[Character]"
  Line 188: Argument of type "object" cannot be assigned to parameter "current_session" of type "AsyncSession" in function "_fetch_character_logic"
  Line 205: "get_base_stats_for_new_character" is not a known attribute of "None"
  Line 205: Cannot access attribute "get_base_stats_for_new_character" for class "RuleEngine"
  Line 208: "get_default_starting_location" is not a known attribute of "None"
  Line 208: Cannot access attribute "get_default_starting_location" for class "LocationManager"
  Line 214: "get_default_bot_language" is not a known attribute of "None"
  Line 214: Cannot access attribute "get_default_bot_language" for class "GameManager"
  Line 229: Cannot access attribute "to_db_dict" for class "Character"
  Line 244: Cannot access attribute "from_db_model" for class "type[Character]"
  Line 265: Cannot access attribute "begin" for class "object"
  Line 267: Argument of type "object" cannot be assigned to parameter "db_session" of type "AsyncSession" in function "get_entity_by_attributes"
  Line 272: Argument of type "object" cannot be assigned to parameter "active_db_session" of type "AsyncSession" in function "_create_new_character_core_logic"
  Line 367: Cannot access attribute "to_db_dict" for class "Character"
  Line 416: Cannot access attribute "from_db_model" for class "type[Character]"

[bot\game\npc_processors\npc_action_processor.py] - 18 issue(s)
  Line 188: Argument missing for parameter "npc_id"
  Line 203: Cannot access attribute "is_busy" for class "NpcManager"
  Line 226: Argument missing for parameter "npc_id"
  Line 270: Argument missing for parameter "npc_id"
  Line 315: Expected 1 positional argument
  Line 535: Cannot access attribute "can_craft" for class "CraftingManager"
  Line 682: Argument missing for parameter "guild_id"
  Line 687: Argument missing for parameter "guild_id"
  Line 694: Argument missing for parameter "guild_id"
  Line 726: Argument missing for parameter "npc_id"
  Line 888: Argument missing for parameter "npc_id"
  Line 988: Argument missing for parameter "npc_id"
  Line 999: Argument of type "Any | None" cannot be assigned to parameter "action_type" of type "str" in function "get_handler"
  Line 999: Type "BaseNpcActionHandler | None" is not assignable to declared type "NpcActionHandlerRegistry | None"
  Line 1009: Cannot access attribute "handle" for class "NpcActionHandlerRegistry"
  Line 1070: Argument missing for parameter "npc_id"
  Line 1118: Argument missing for parameter "entity_id"
  Line 1118: No parameter named "context"

[bot\command_modules\quest_cmds.py] - 17 issue(s)
  Line 48: "db_service" is not a known attribute of "None"
  Line 57: Argument of type "object" cannot be assigned to parameter "db_session" of type "AsyncSession" in function "get_entity_by_attributes"
  Line 89: "get_rule" is not a known attribute of "None"
  Line 110: Argument of type "object" cannot be assigned to parameter "db_session" of type "AsyncSession" in function "get_entity_by_id"
  Line 111: Argument of type "object" cannot be assigned to parameter "db_session" of type "AsyncSession" in function "get_entity_by_id"
  Line 114: Argument of type "Column[Unknown]" cannot be assigned to parameter "data_i18n" of type "Dict[str, Any]" in function "_get_i18n_value"
  Line 114: Argument of type "str | Any | None" cannot be assigned to parameter "lang" of type "str" in function "_get_i18n_value"
  Line 115: Invalid conditional operand of type "Column[Unknown]"
  Line 117: Argument of type "Column[Unknown]" cannot be assigned to parameter "data_i18n" of type "Dict[str, Any]" in function "_get_i18n_value"
  Line 117: Argument of type "str | Any | None" cannot be assigned to parameter "lang" of type "str" in function "_get_i18n_value"
  Line 118: Invalid conditional operand of type "Column[Unknown]"
  Line 120: Argument of type "Column[Unknown]" cannot be assigned to parameter "data_i18n" of type "Dict[str, Any]" in function "_get_i18n_value"
  Line 120: Argument of type "str | Any | None" cannot be assigned to parameter "lang" of type "str" in function "_get_i18n_value"
  Line 121: Invalid conditional operand of type "Column[Unknown]"
  Line 131: Invalid conditional operand of type "Column[Unknown]"
  Line 132: Argument of type "Column[Unknown]" cannot be assigned to parameter "data_i18n" of type "Dict[str, Any]" in function "_get_i18n_value"
  Line 132: Argument of type "str | Any | None" cannot be assigned to parameter "lang" of type "str" in function "_get_i18n_value"

[bot\game\services\location_interaction_service.py] - 17 issue(s)
  Line 56: Cannot access attribute "get" for class "object"
  Line 63: Cannot access attribute "get" for class "object"
  Line 65: Cannot access attribute "get" for class "object"
  Line 101: Expected 1 positional argument
  Line 110: "Dict[str, Any]" is not awaitable
  Line 110: "None" is not awaitable
  Line 158: Cannot access attribute "get_rules_config" for class "RuleEngine"
  Line 183: Invalid conditional operand of type "Column[Unknown]"
  Line 196: Invalid conditional operand of type "Column[Unknown] | bool"
  Line 203: Type "tuple[Literal[True], Unknown | ColumnElement[Any] | None]" is not assignable to return type "Tuple[bool, str]"
  Line 214: Cannot access attribute "character_has_item_template" for class "InventoryManager"
  Line 237: Cannot access attribute "send_character_feedback" for class "NotificationService"
  Line 268: Argument missing for parameter "consequences"
  Line 270: No parameter named "consequences_list"
  Line 271: No parameter named "context"
  Line 272: No parameter named "actor_id"
  Line 273: No parameter named "actor_type"

[tests\cogs\test_master_commands.py] - 17 issue(s)
  Line 75: "db_service" is not a known attribute of "None"
  Line 75: "get_session_factory" is not a known attribute of "None"
  Line 75: Cannot assign to attribute "return_value" for class "FunctionType"
  Line 95: Argument of type "Column[str]" cannot be assigned to parameter "source_location_id" of type "str"
  Line 95: Parameter "source_location_id" is already assigned
  Line 96: Argument of type "Column[str]" cannot be assigned to parameter "target_location_id" of type "str"
  Line 105: Argument of type "Column[Unknown]" cannot be assigned to parameter "obj" of type "Sized" in function "len"
  Line 107: Invalid conditional operand of type "ColumnElement[bool]"
  Line 108: Invalid conditional operand of type "ColumnElement[bool]"
  Line 109: Invalid conditional operand of type "ColumnElement[bool]"
  Line 115: Cannot access attribute "assert_awaited_once" for class "FunctionType"
  Line 116: Cannot access attribute "call_args" for class "FunctionType"
  Line 129: "db_service" is not a known attribute of "None"
  Line 129: "get_session_factory" is not a known attribute of "None"
  Line 129: Cannot assign to attribute "return_value" for class "FunctionType"
  Line 135: Parameter "source_location_id" is already assigned
  Line 138: Cannot access attribute "assert_awaited_once_with" for class "FunctionType"

[bot\game\command_handlers\action_commands.py] - 16 issue(s)
  Line 85: "get_character_by_discord_id" is not a known attribute of "None"
  Line 96: "get_npc" is not a known attribute of "None"
  Line 99: "get_npc_by_name" is not a known attribute of "None"
  Line 109: "get_location_name" is not a known attribute of "None"
  Line 110: "get_location_name" is not a known attribute of "None"
  Line 114: "get_combat_by_participant_id" is not a known attribute of "None"
  Line 117: "get_combat_by_participant_id" is not a known attribute of "None"
  Line 122: "start_combat" is not a known attribute of "None"
  Line 210: "get_character_by_discord_id" is not a known attribute of "None"
  Line 215: "get_npc" is not a known attribute of "None"
  Line 218: "get_npc_by_name" is not a known attribute of "None"
  Line 235: "process_steal_action" is not a known attribute of "None"
  Line 282: "get_character_by_discord_id" is not a known attribute of "None"
  Line 292: "get_npc" is not a known attribute of "None"
  Line 294: "get_character" is not a known attribute of "None"
  Line 300: "process_use_item_action" is not a known attribute of "None"

[bot\game\managers\game_manager.py] - 16 issue(s)
  Line 26: Import "bot.database.models.pending_generation" could not be resolved
  Line 281: Cannot assign to attribute "combat_manager" for class "PartyManager"
  Line 285: Cannot assign to attribute "dialogue_manager" for class "NpcManager"
  Line 289: Argument of type "None" cannot be assigned to parameter "quest_manager" of type "QuestManager" in function "__init__"
  Line 296: Cannot assign to attribute "quest_manager" for class "ConsequenceProcessor"
  Line 297: Cannot assign to attribute "quest_manager" for class "DialogueManager"
  Line 301: Cannot access attribute "notification_service" for class "DialogueManager"
  Line 302: Cannot access attribute "notification_service" for class "QuestManager"
  Line 303: Cannot access attribute "notification_service" for class "ConsequenceProcessor"
  Line 357: Argument of type "None" cannot be assigned to parameter "event_stage_processor" of type "EventStageProcessor" in function "__init__"
  Line 359: Cannot assign to attribute "event_stage_processor" for class "EventActionProcessor"
  Line 395: Argument of type "OpenAIService | None" cannot be assigned to parameter "openai_service" of type "OpenAIService" in function "__init__"
  Line 455: Cannot assign to attribute "multilingual_prompt_generator" for class "QuestManager"
  Line 456: Cannot assign to attribute "prompt_context_collector" for class "ConsequenceProcessor"
  Line 459: Cannot assign to attribute "ai_validator" for class "QuestManager"
  Line 771: Cannot access attribute "request_content_generation" for class "AIGenerationService"

[bot\game\rules\action_processor.py] - 16 issue(s)
  Line 85: Cannot access attribute "get_location" for class "LocationManager"
  Line 102: Cannot access attribute "get_active_events_in_location" for class "EventManager"
  Line 111: Cannot access attribute "name" for class "CoroutineType[Any, Any, Character | None]"
  Line 117: Cannot access attribute "id" for class "CoroutineType[Any, Any, Character | None]"
  Line 136: Cannot access attribute "name" for class "CoroutineType[Any, Any, Character | None]"
  Line 146: Cannot access attribute "name" for class "CoroutineType[Any, Any, Character | None]"
  Line 150: Cannot access attribute "id" for class "CoroutineType[Any, Any, Character | None]"
  Line 168: Cannot access attribute "get_exit_target" for class "LocationManager"
  Line 173: Cannot access attribute "find_location_by_name_or_id" for class "LocationManager"
  Line 202: Cannot access attribute "id" for class "CoroutineType[Any, Any, Character | None]"
  Line 210: Cannot access attribute "name" for class "CoroutineType[Any, Any, Character | None]"
  Line 252: Cannot access attribute "skills" for class "CoroutineType[Any, Any, Character | None]"
  Line 260: Cannot access attribute "perform_check" for class "RuleEngine"
  Line 274: Cannot access attribute "name" for class "CoroutineType[Any, Any, Character | None]"
  Line 274: Cannot access attribute "skills" for class "CoroutineType[Any, Any, Character | None]"
  Line 274: Cannot access attribute "stats" for class "CoroutineType[Any, Any, Character | None]"
